src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2813,6 +2814,30 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Moves the tab to another index in the tabs row (if required).
+    // Arguments:
+    // - currentTabIndex: the current index of the tab to move
+    // - suggestedNewTabIndex: the new index of the tab, might get clamped to fit int the tabs row boundaries
+    // Return Value:
+    // - <none>
+    void TerminalPage::_TryMoveTab(const uint32_t currentTabIndex, const int32_t suggestedNewTabIndex)
+    {
+        uint32_t newTabIndex = std::clamp<int32_t>(suggestedNewTabIndex, 0, _tabs.Size() - 1);
```

Comment:
qq: should tab move wrap around like tab switch? moving to the right when you're at the end _might_ want to move to the far left?

Reply:
Sincerely speaking I am not sure what will be more useful, though this will be more aligned with what we have when dragging with a mouse.